src/yaml-support/yaml-schema.ts

```diff
@@ -0,0 +1,165 @@
+import * as _ from 'lodash';
+import Uri from 'vscode-uri';
+import * as vscode from 'vscode';
+import { yamlLocator } from "./yaml-locator";
+import { VSCODE_YAML, KUBERNETES_SCHEMA, KUBERNETES_GROUP_VERSION_KIND, GROUP_VERSION_KIND_SEPARATOR } from "./yaml-constant";
+
+// the schema for kubernetes
+const _definitions = {};
+
+// load the kubernetes schema and make some modifications to $ref node
+export function loadSchema(schemaFile: string) {
+    const schemaRaw = require(schemaFile);
+    const definitions = schemaRaw.definitions;
+    const toFixed = [];
+    for (let key of Object.keys(definitions)) {
+        let node = definitions[key];
+        const name = key;
+        if (node[KUBERNETES_GROUP_VERSION_KIND]) {
+            let groupKindNode = node[KUBERNETES_GROUP_VERSION_KIND];
+            delete node[KUBERNETES_GROUP_VERSION_KIND];
+            if (groupKindNode.length) {
+                groupKindNode.forEach((groupKindNodeItem) => {
+                    const apiVersion = ((groupKindNodeItem.Group ? groupKindNodeItem.Group + '/' : '') + groupKindNodeItem.Version);
+                    const kind = groupKindNodeItem.Kind;
+                    _definitions[apiVersion + GROUP_VERSION_KIND_SEPARATOR + kind] = _definitions[name.toLowerCase()] = {
+                        apiVersion,
+                        name,
+                        kind,
+                        ...node
+                    };
+
+
+                });
+            }
+        } else {
+            _definitions[name.toLowerCase()] = {
+                name,
+                ...node
+            };
+        }
+
+        toFixed.push(node['properties']);
+    }
+
+    for (let fix of toFixed ) {
+        patchOnRef(fix);
+    }
+}
+
+export async function registerYamlSchemaSupport() {
+    loadSchema(KUBERNETES_SCHEMA);
+    let yamlPlugin;
+    const installedExtensions: any[] = vscode.extensions.all;
+    for (let ext of installedExtensions) {
+        if (ext.id === VSCODE_YAML) {
+            try {
+                yamlPlugin = await Promise.resolve(ext.activate());
+                break;
+            } catch (error) {
+                console.log('Failed to activate the Yaml Extension: ' + error);
+            }
+            break;
+        }
+    }
+    if (!yamlPlugin) {
+        console.log('vscode-yaml plugin doesn\'t expose the entry to register custom schema provider.');
```

Comment:
What does this mean?  The user won't see this message so this will silently return `undefined`... what is the impact of that?

Reply:
The old version of vscode-yaml plugin without my custom schema code will not expose the register... apis, just ignore it and the yaml schema will not be supported.

Reply:
It sounds like this would result in the user not getting the schema support, and not knowing why.  I feel we should either declare a package dependency with minimum version, or display a warning explaining why we're not providing (intellisense / schema validation / whatever the schema is giving us) and how they can fix it.